BGAnimations/ScreenSystemLayer overlay.lua

```diff
@@ -519,27 +519,40 @@ t[#t+1] = Def.ActorFrame {
 		self.IsDisplaying = false
 	end,
 	SystemMessageMessageCommand=function(self, params)
-		if self.IsDisplaying then
-			self:finishtweening()
-			local newText = bmt:GetText().."\n"..params.Message
-			-- Display only the last few lines of text
-			local lines = {}
-			for line in newText:gmatch("[^\n]+") do
-				lines[#lines+1] = line
+		-- Handle case where the message usage is SM(msg, duration)
+		local stack = params.Stack or false
+		if type(stack) == "number" then
+			duration = stack
+		end
+
+		if stack then
+			if self.IsDisplaying then
+				self:finishtweening()
+				local newText = bmt:GetText().."\n"..params.Message
+				-- Display only the last few lines of text
+				local lines = {}
+				for line in newText:gmatch("[^\n]+") do
+					lines[#lines+1] = line
+				end
+				local start = math.max(#lines - totalVisibleLines, 1)
+				local displayText = table.concat(lines, "\n", start, #lines)
+				bmt:settext(displayText)
+			else
+				bmt:settext( params.Message )
+			end
+			self:playcommand( "On")

```

Comment:
```lua
      self:playcommand( "On")
			if params.NoAnimate then
				self:finishtweening()
			end
			self:sleep(type(params.Duration)=="number" and params.Duration or 3.33 + 0.25):queuecommand("Off")
```

Should be common to both the stack and not stacked branch. We can move it out and put it below the condition.